### ISSUE TITLE
Report invalid index

### ISSUE THREAD

**Issue by gajus** [benmosher/eslint-plugin-import#563]
It is a common pattern to have an index file that exports all the siblings, e.g.

Suppose the following directory structure.

```
./src/loaders/
├── BadgeByBadgeIdLoader.js
├── BadgesByPostIdLoader.js
├── BlogByBlogIdLoader.js
├── PostsByBlogIdLoader.js
├── UserByUserIdLoader.js
└── index.js
```

Then the `./index.js` would contain:

```js
export BadgeByBadgeIdLoader from './BadgeByBadgeIdLoader';
export BadgesByPostIdLoader from './BadgesByPostIdLoader';
export BlogByBlogIdLoader from './BlogByBlogIdLoader';
export PostsByBlogIdLoader from './PostsByBlogIdLoader';
export UserByUserIdLoader from './UserByUserIdLoader';
```

This rule would check all `./index.js` files that only have import/export statements.

If a matched file does not list one of the sibling files in the directory, it would report a problem.

**Comment by benmosher** [comment]
I like this. I think it could check every `index.js` and let folks add `/* eslint-disable import/[name tbd] */` upfront if the file doesn't match? I like the idea of enforcing that _every_ index file exports _something_ from every sibling in the package.

Any helper files would then need to be in a subfolder, does that work? and the subfolder (even if it has an `index.js`) would not be enforce to have exports in the ancestor's `index.js`?

**Comment by m-nathani** [comment]
Hii, i still gettings error on import... where as i have a file in FeatureFlag/index.tsx

```bash
  10:25  error  Missing file extension for "../FeatureFlag"       import/extensions
```

with eslintrc has this rule too

```json
    "import/extensions": [
      2,
      "ignorePackages",
      {
        "js": "never",
        "jsx": "never",
        "ts": "never",
        "tsx": "never",
      },
    ],
```

does anyone know how to resolve it ?

**Comment by ljharb** [comment]
@m-nathani do you have a `FeatureFlag.something` file as well? or just a `FeatureFlag` dir with an index file inside it?

**Comment by m-nathani** [comment]
Just index .. nothing else

**Comment by ljharb** [comment]
@m-nathani a repro repo on a new issue would be helpful; that shouldn't be warning. Do make sure that you have the import/extensions *setting* defined also to include `tsx`.

**Comment by soryy708** [comment]
This complements: https://github.com/import-js/eslint-plugin-import/issues/2922
